bagel/utilities/pheno_utils.py

```diff
@@ -77,6 +81,53 @@ def get_annotated_columns(data_dict: dict) -> list(tuple[str, dict]):
     ]
 
 
+def recursive_find_values_for_key(data: dict, target: str) -> list:
+    """
+    Recursively search for a key in a possibly nested dictionary and return a list of all values found for that key.
+
+    TODO: This function currently only considers nested dicts, and would need to be expanded if Neurobagel
+    data dictionaries grow to have controlled terms inside list objects.
+    """
+    target_values = []
+    if isinstance(data, dict):
+        for key, value in data.items():
+            if key == target:
+                target_values.append(value)
+            else:
+                target_values.extend(
+                    recursive_find_values_for_key(data=value, target=target)
+                )
+    return target_values
+
+
+def find_unsupported_namespaces_and_term_urls(
+    data_dict: dict,
+) -> tuple[list, dict]:
+    """
+    From a provided data dictionary, find all term URLs that contain an unsupported namespace prefix.
+    Return a tuple of unsupported prefixes and a dictionary of the offending column names and their unrecognized term URLs.
+    """
+    unsupported_prefixes = set()
+    unrecognized_term_urls = {}
+
+    for col, content in get_annotated_columns(data_dict):
+        for col_term_url in recursive_find_values_for_key(
+            content["Annotations"], "TermURL"
```

Comment:
```suggestion
            content.get("Annotations", {}), "TermURL"
```
:cherries: but unless we're super sure this happens after validation that checks presence of "Annotations" it'd be nicer to crash out here I think

Reply:
Yeah, `get_annotated_columns()` should only return columns containing the "Annotations" key (and it has a test), so I think it may be better to just break here if for some reason it behaves unexpectedly 🧐